hamilton/execution/executors.py

```diff
@@ -396,20 +407,32 @@ def run_graph_to_completion(
                             f"{[item.name for item in next_task.nodes]}"
                         )
                         raise e
-                    task_futures[next_task.task_id] = submitted
```

Comment:
Why the change here? Assuming `Task` is hashable? Don't see it in the spec...

Reply:
Oh, see you changed it... Got it, this allows us to have the extra data available...

Reply:
Yup that was what I was going for! I needed to access the `adapater` for the task and thought adding a parallel structure was a little redundent. Do you see any problems with making `TaskImplementation` hashable?